cctray_xml_feed_mock/src/server.js

```diff
@@ -19,6 +19,28 @@ function response(file) {
   }
 }
 
+const sleep = (waitTimeInMs) => new Promise(resolve => setTimeout(resolve, waitTimeInMs));
+
+function delayedResponse(file) {
+  return async function (req, res) {
+    req.connection.on('close',function(){
+      console.log("Connection closed by client")
+    });
+
+    fs.readFile('resources/' + file, 'utf8', async function (err, contents) {
+      const lines = contents.split("\n")
+      const sleepTime = Math.ceil((req.query.delay || 0) / lines.length)
+      res.setHeader('Content-Type', 'application/xml; charset=utf-8')
+      for (let i = 0 ; i < lines.length; i++) {
+        const line = lines[i];
+        await sleep(sleepTime)
+        res.write(line)
+      }
+      res.end('')
+    })
+  }
+}
```

Comment:
Simulating slow data transfer (line by line) by streaming data after a calculated delay from `query` parameter in the call.
Ex: Below given call will respond after 5 seconds
http://localhost:5050/slow/go/cctray.xml/?delay=5000